pkgs/development/compilers/gcc/9/default.nix

```diff
@@ -192,7 +194,9 @@ stdenv.mkDerivation ({
 
   depsBuildBuild = [ buildPackages.stdenv.cc ];
   nativeBuildInputs = [ texinfo which gettext ]
-    ++ (optional (perl != null) perl);
+    ++ (optional (perl != null) perl)
+    ++ (optional (expect != null) expect)
+    ++ (optional (expect != null) dejagnu);
```

Comment:
should be dejagnu instead of expect. Maybe you could include them only when doCheck = true instead ? or add an assert to make sure expect/dejagnu are not null when doCheck is true.

Reply:
Heh, copy & paste fail. I would prefer if it was gated on `doCheck`, although I'm worried that if I just do that then afterwards use `.override({...}: { doCheck = true; })` it won't add those dependencies after the fact (I'm not sure how lazy these fields are tbh).

Reply:
as @teto suggested, add asserts and check against doCheck to append.